runtime/vm/resolvesupport.cpp

```diff
@@ -872,8 +872,32 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 				targetClass = currentTargetClass;
 				goto illegalAccess;
 			}
+			
+			/* TODO: Redo value class check when L-World spec is finalized */
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			/* When setting a field, throw IncompatibleClassChangeError if the flag
+			 * J9_RESOLVE_FLAG_CHECK_VALUE_CLASS is set but resolvedClass is not a value type
+			 * or if J9_RESOLVE_FLAG_CHECK_VALUE_CLASS is not set and the resolved class is a
+			 * value type.
+			 */
+			if (J9_ARE_ALL_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_FIELD_SETTER)
+				&& (J9_ARE_ALL_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_CHECK_VALUE_CLASS)
+					!= J9_ARE_ALL_BITS_SET(resolvedClass->romClass->modifiers, J9AccValueType))
```

Comment:
This check bothers me and I don't have a good reason why....  comparing the results of ALL_BITS to each other feels wrong.  Can you express it differently?